src/api/route-services/access/index.js

```diff
@@ -58,6 +58,61 @@ class AccessService {
 
     const emailBody = buildUserInvitedEmailBody(userEmail, experimentId, inviterUser);
     await sendEmail(emailBody);
+
+    return OK();
+  }
+
+  async revokeRole(userEmail, experimentId) {
+    logger.log('Removing role for user ', userEmail, ' in experiment ', experimentId);
+
+    const userAttributes = await getAwsUserAttributesByEmail(userEmail);
+    const userId = userAttributes.find((attr) => attr.Name === 'sub').Value;
+
+    const key = convertToDynamoDbRecord({
+      experimentId,
+      userId,
+    });
+    const dynamoParams = {
+      TableName: this.userAccessTableName,
+      Key: key,
+    };
+
+    const dynamodb = createDynamoDbInstance();
+
+    try {
+      await dynamodb.deleteItem(dynamoParams).send();
+      return OK();
+    } catch (e) {
+      if (e.statusCode === 404) throw new NotFoundError(`Role for user ${userEmail} in experiment ${experimentId} not found`);
+      throw e;
+    }
+  }
+
+  async getRoles(experimentId) {
+    logger.log('Getting access for experiment ', experimentId);
+    const experimentEntries = await this.getExperimentEntries(
```

Comment:
The `getRoles` should also add the information in `invite-access`. However, we need to show that those are pending invitations. I can think of 2 options:
1. For the `invite-access` entries, change the role to "Pending" so the UI will automatically list those as pending
2. Add an extra field to each entry response (maybe `pending=bool`). Then in the UI show that those with the pending flag (or whatever field) are indeed pending.